internal/operators/ocs/ocs_operator.go

```diff
@@ -92,6 +94,43 @@ func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
 }
 
 // GetHostRequirements provides operator's requirements towards the host
-func (o *operator) GetHostRequirements(context.Context, *common.Cluster, *models.Host) (*models.ClusterHostRequirementsDetails, error) {
-	return &models.ClusterHostRequirementsDetails{}, nil
+func (o *operator) GetHostRequirements(_ context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	numOfHosts := len(cluster.Hosts)
+
+	if numOfHosts < 3 {
+		return nil, fmt.Errorf("number of hosts is less than 3")
+	} else if numOfHosts == 3 { // Compact Mode
+		if host.Role == models.HostRoleMaster || host.Role == models.HostRoleAutoAssign {
+			return &models.ClusterHostRequirementsDetails{
+				CPUCores:   CPUCompactMode,
+				RAMMib:     MemoryCompactMode,
+				DiskSizeGb: MinDiskSize,
+			}, nil
+		}
+		return nil, fmt.Errorf("unsupported role: %s", host.Role)
+	} else if host.Role == models.HostRoleAutoAssign { //If auto-assign we cannot proceed with minimal or standard mode
```

Comment:
No need for keep doing `else`, since you already return with error inside the previous `if`.